pkg/credential/token.go

```diff
@@ -30,12 +35,49 @@ type GcpTokenFetcher struct {
 	Aud string
 }
 
+// AwsTokenFetcher implements the token fetcher in AWS.
+// It gets the EC2 instance identity document as described
+// in the doc: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+type AwsTokenFetcher struct {
```

Comment:
Can we rename all TokenFetchers to IdentityDocFethers, to be more precise?

Reply:
I think that can go with a follow-up PR. That will make this PR clean and easier to track :)

Reply:
Yes, that works :)